src/lib/optimizer/strategy/projection_combination_rule.cpp

```diff
@@ -0,0 +1,106 @@
+#include "projection_combination_rule.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+
+namespace opossum {
+
+std::string ProjectionCombinationRule::name() const { return "Projection Combination Rule"; }
+
+bool ProjectionCombinationRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) {
+  if (node->type() != LQPNodeType::Projection) {
+    return _apply_to_children(node);
+  }
+
+  std::vector<std::shared_ptr<ProjectionNode>> projection_nodes;
+
+  // Gather subsequent ProjectionNodes
+  auto current_node = node;
+  while (current_node->type() == LQPNodeType::Projection) {
+    projection_nodes.emplace_back(std::static_pointer_cast<ProjectionNode>(current_node));
```

Comment:
So you are emplacing a node that has potentially multiple parents? Isn't that what you didn't want to do?

Reply:
oh wait, so the "root ProjectionNode" is the node closest to the leaves of the LQP? That's a bit of a confusing terminology ;)

Reply:
Ah yes, I was haggling with the wording there. I rephrased the commenting a little bit, hoping that makes it clearer:
```
/**
 * This optimizer rule combines consecutive ProjectionNodes if they all have exactly one parent
 * (except for the node closest to the tree root in a sequence of ProjectionNodes, which can have multiple parents).
 *
 * The rule removes all but the ProjectionNode closest to the tree root (the top node of that sequence) from the tree.
 * If the top ProjectionNode contains any LQPColumnReferences to a column created by one of the ProjectionNodes
 * to be removed, then the LQPColumnReference is replaced by the actual column.
 */
```

Reply:
"* The rule removes all but the ProjectionNode closest to the tree root" technically, this one is removed as well. How about "If possible, replaces a chain of ProjectionNodes with a single ProjectionNode."